driver-4/src/test/java/com/azure/cosmos/cassandra/TestCommon.java

```diff
@@ -36,33 +40,39 @@ private TestCommon() {
 
     private static final Pattern HOSTNAME_AND_PORT = Pattern.compile("^\\s*(?<hostname>.*?):(?<port>\\d+)\\s*$");
 
-    static final List<String> CONTACT_POINTS = Arrays.asList(getPropertyOrEnvironmentVariable(
-        "azure.cosmos.cassandra.contact-point",
-        "AZURE_COSMOS_CASSANDRA_CONTACT_POINT",
-        "localhost:10350").split("\\s*,\\s*"));
-
-    static final String GLOBAL_ENDPOINT = getPropertyOrEnvironmentVariable(
+    static final InetSocketAddress GLOBAL_ENDPOINT = parseSocketAddress(getPropertyOrEnvironmentVariable(
         "azure.cosmos.cassandra.global-endpoint",
         "AZURE_COSMOS_CASSANDRA_GLOBAL_ENDPOINT",
-        "localhost:10350");
-
-    static final List<InetSocketAddress> NODES = Arrays.stream(getPropertyOrEnvironmentVariable(
-        "azure.cosmos.cassandra.nodes",
-        "AZURE_COSMOS_CASSANDRA_NODES",
-        "localhost:10350").split("\\s*,\\s*"))
-        .map(value -> {
-            final Matcher match = matchSocketAddress(value);
-            return new InetSocketAddress(match.group("hostname"), Integer.parseUnsignedInt(match.group("port")));
-        }).collect(Collectors.toList());
-
-    static final String PASSWORD = getPropertyOrEnvironmentVariable(
-        "azure.cosmos.cassandra.password",
-        "AZURE_COSMOS_CASSANDRA_PASSWORD",
-        "");
-    static final String USERNAME = getPropertyOrEnvironmentVariable(
-        "azure.cosmos.cassandra.username",
-        "AZURE_COSMOS_CASSANDRA_USERNAME",
-        "");
+        "localhost:10350"));
+
+    static final List<String> PREFERRED_REGIONS = getPropertyOrEnvironmentVariableList(
+        "azure.cosmos.cassandra.preferred-region-",
+        "AZURE_COSMOS_CASSANDRA_PREFERRED_REGION_",
+        5);
+
+    static final List<SocketAddress> REGIONAL_ENDPOINTS = Arrays.stream(getPropertyOrEnvironmentVariable(
+        "azure.cosmos.cassandra.regional-endpoints",
```

Comment:
How is the regional endpoints used in conjunction with the preferred region.

Reply:
If above is true, I see the Vivek's point that the settings could be confusing. Will this work:
Only provide preferred-regions and multi-region-writes settings. Remove read-datacenter and write-datacenter.
When multi-region-writes is false which is the default value
    writes go to the global endpoint, aka contact points, aka system.local
    reads honor preferred-regions list if it's provided; otherwise, behave the same as writes
When multi-region-writes is true
    both reads and writes honor preferred-region list if it's provided; otherwise are routed to global endpoint.